packages/pieces/google-sheets/src/lib/actions/insert-row.action.ts

```diff
@@ -16,11 +15,45 @@ export const insertRowAction = createAction({
             description: 'Inserted values that are dates and formulas will be entered strings and have no effect',
             required: false,
         }),
-        values: Property.Array({
+        values: Property.DynamicProperties({
             displayName: 'Values',
-            description: 'These are the cell values of the row that will be added, beginning with the Value in column A and proceeding with each Value being entered in the next cell.',
+            description: 'The values to insert',
             required: true,
-        }),
+            refreshers: ['authentication', 'sheet_id', 'spreadsheet_id'],
+            props: async (context) => {
+                
+                const authentication = context.authentication as OAuth2PropertyValue;
+                const spreadsheet_id = context.spreadsheet_id as unknown as string;
+                const sheet_id = context.sheet_id as unknown as number;
+                const accessToken = authentication['access_token'] ?? '';
+
+                const sheetName = await googleSheetsCommon.findSheetName(accessToken, spreadsheet_id, sheet_id);
+
+                if (!sheetName) {
+                    throw Error("Sheet not found in spreadsheet");
+                }
+
+                const values = await googleSheetsCommon.getValues(spreadsheet_id, accessToken, sheet_id);
+
+                
+                const firstRow = values[0].values;
```

Comment:
Same Issue here, if there is no rows this will throw error I think we should show default option like columns names.